### PR TITLE
feat(FOROME-1254): Added preset and dtree to URL

### PR DESCRIPTION
**Pull request by AndreyUstyumenko**
https://quantori.atlassian.net/browse/FOROME-1254

### CODE REVIEW

File: src/store/filter/filter.store.ts [hunk 1]
```diff
@@ -15,6 +15,7 @@ import {
 import { IDsListArguments } from '@service-providers/dataset-level'
 import filteringRegimeProvider from '@service-providers/filtering-regime/filtering-regime.provider'
 import { showToast } from '@utils/notifications'
+import { updateURLParams } from './../../utils/url/update-url-params'
```
Comment (EvgeniySobolev) [inline src/store/filter/filter.store.ts]:
```suggestion
import { updateURLParams } from '@utils/url/update-url-params'
```

**Review by EvgeniySobolev — APPROVED** [review]

File: src/utils/url/update-url-params.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export const updateURLParams = (paramName: string, data: any) => {
```
Comment (AlMaQntr) [inline src/utils/url/update-url-params.ts]: i created @src/core/history.ts for same purposes)

File: src/pages/filter/refiner/refiner.page.tsx [hunk 3]
```diff
@@ -29,10 +33,12 @@ export const RefinerPage = observer((): ReactElement => {
   useDatasetName()
 
   useEffect(() => {
+    presetName && applyPreset(presetName)
```
Comment (AlMaQntr) [inline src/pages/filter/refiner/refiner.page.tsx:36]: For variant drawer i moved similar logic to the store. see `variantStore.observeVarianthistory`
It doesn't matter, but may be, we can discuss general style for history control

**Review by AlMaQntr — APPROVED** [review]

**Review by Lapk1n — APPROVED** [review]

File: src/pages/filter/refiner/refiner.page.tsx (hunk 3 above)
Comment (AndreyUstyumenko) [reply src/pages/filter/refiner/refiner.page.tsx:36]: Decided to rework this within a separate task

File: src/utils/url/update-url-params.ts (hunk 2 above)
Comment (AndreyUstyumenko) [reply src/utils/url/update-url-params.ts]: done

**Pull request merged**